product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md

```diff
@@ -0,0 +1,178 @@
+# 1.6.1 Modify payload by adding a child element using enrich mediator
+
+## When to use
+The Enrich Mediator can process a message based on a given source configuration and then perform the specified action on the message by using the target configuration. It is often used to do slight modifications to the payload. There are main three target actions as “Replace”, “Child” and “Sibling”. In this scenario we focus on adding a child element to the payload. 
+
+
+## Sample use case
+Assume that we have our original payload to be sent as below. 
+
+original payload
+```
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ser="http://services.samples" xmlns:xsd="http://services.samples/xsd">
+<soapenv:Header/>
+<soapenv:Body>
+<ser:getQuote>
+<!--Optional:-->
+<ser:request>
+<!--Optional:-->
+<xsd:symbol>WSO2</xsd:symbol>
+</ser:request>
+</ser:getQuote>
+</soapenv:Body>
+</soapenv:Envelope>
+```
+
+Before sending this to the endpoind, we need to modify this payload by adding a child element to ```<ser:request>``` element. The child element will be ```<xsd:symbol>SUN</xsd:symbol>```
+
+Required payload
+```
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ser="http://services.samples" xmlns:xsd="http://services.samples/xsd">
+<soapenv:Header/>
+<soapenv:Body>
+<ser:getQuote>
+<!--Optional:-->
+<ser:request>
+<!--Optional:-->
+<xsd:symbol>WSO2</xsd:symbol>
+<xsd:symbol>SUN</xsd:symbol>
+</ser:request>
+</ser:getQuote>
+</soapenv:Body>
+</soapenv:Envelope>
+```
+
+This slight modification to the payload can be done using enrich mediator. 
+
+mediator
+```
+<enrich xmlns="http://ws.apache.org/ns/synapse">
+    <source clone="true" type="inline">
+        <xsd:symbol xmlns:xsd="http://services.samples/xsd">SUN</xsd:symbol>
+    </source>
+    <target action="child" type="custom"
+        xmlns:ser="http://services.samples"
+        xmlns:xsd="http://services.samples/xsd" xpath="//ser:getQuote/ser:request"/>
+</enrich>
+```
+
+## Prerequisites
+SimpleStockQuote Service needs to have deployed. 
+
+## Development guidelines
+
+Sequence and proxy Configuration
+
+Create the following sequence and add to the proxy service. 
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy xmlns="http://ws.apache.org/ns/synapse"
+       name="Enrich2"
+       startOnLoad="true"
+       statistics="disable"
+       trace="disable"
+       transports="http,https">
+   <target>
+      <inSequence>
+         <log level="full"/>
+         <enrich>
+            <source clone="true" type="inline">
+               <xsd:symbol xmlns:xsd="http://services.samples/xsd">SUN</xsd:symbol>
+            </source>
+            <target xmlns:xsd="http://services.samples/xsd"
+                    xmlns:ser="http://services.samples"
+                    action="child"
+                    xpath="//ser:getQuote/ser:request"/>
+         </enrich>
+         <log level="full"/>
```

Comment:
logging is not the best way to demonstrate samples. let's respond the enriched message

Reply:
This is updated to use a respond mediator. Therefore removed the logging.